src/server/JSON/Parser.cpp

```diff
@@ -0,0 +1,257 @@
+/**
+ * @file Parser.cpp
+ * @brief Implementation of the Parser class.
+ *
+ * @details
+ * This file implements the methods for parsing JSON content from a file.
+ * It supports parsing of strings, numbers, booleans, null, arrays, and objects.
+ */
+
+#include "Parser.hpp"
+
+#include <cctype>
+#include <filesystem>
+#include <fstream>
+#include <stdexcept>
+
+namespace ISXJson
+{
+Parser::Parser(const std::filesystem::path& file_path)
+{
+	std::ifstream file(file_path);
+	if (!file) throw std::runtime_error("Could not open file: " + file_path.string());
+
+	m_input = std::string((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+	m_pos = 0;
+}
+
+void Parser::SkipWhitespace()
+{
+	// Advance m_pos until a non-whitespace character is found.
+	while (m_pos < m_input.size() && (std::isspace(static_cast<unsigned char>(m_input[m_pos])) != 0)) m_pos++;
+}
+
+char Parser::Peek() const
+{
+	// Return the current character without advancing, or '\0' if at the end.
+	return m_pos < m_input.size() ? m_input[m_pos] : '\0';
+}
+
+char Parser::Get()
+{
+	// Ensure that we are not reading past the end of the input.
+	if (m_pos >= m_input.size())
+		throw std::runtime_error("Unexpected end of input at position " + std::to_string(m_pos));
+
+	// Return the character at m_pos and then advance the pointer.
+	return m_input[m_pos++];
+}
+
+JSON Parser::ParseValue()
+{
+	// Skip any whitespace before parsing the JSON value.
+	SkipWhitespace();
+
+	char c = Peek();
+	// Determine type of JSON value based on the current character.
+	switch (c)
+	{
+	case '"': // String value.
+		return ParseString();
+	case '{': // Object value.
+		return ParseObject();
+	case '[': // Array value.
+		return ParseArray();
+	case 't': // Boolean true.
+	case 'f': // Boolean false.
+		return ParseBool();
+	case 'n': // Null value.
+		return ParseNull();
+	default:
+		// If number or negative sign encountered, parse number.
+		if (c == '-' || (std::isdigit(static_cast<unsigned char>(c)) != 0)) return ParseNumber();
+		throw std::runtime_error("Unexpected character: " + std::string(1, c));
+	}
+}
+
+JSON Parser::ParseString()
+{
+	char quote = Get(); // Consume opening quote
+	if (quote != '"') throw std::runtime_error("Expected opening quote for string");
+
+	std::string result;
+	while (true)
+	{
+		if (m_pos >= m_input.size()) throw std::runtime_error("Unterminated string literal");
+		char c = Get();
+		if (c == '"') break;
+		if (c == '\\')
+		{
+			if (m_pos >= m_input.size()) throw std::runtime_error("Unterminated escape sequence");
+			char esc = Get();
+			switch (esc)
+			{
+			case '"':
+				result.push_back('\"');
+				break;
+			case '\\':
+				result.push_back('\\');
+				break;
+			case '/':
+				result.push_back('/');
+				break;
```

Comment:
More compact version
```suggestion
			case '"':
			case '\\':
			case '/':
				result.push_back(esc);
				break;
```